.github/workflows/hello-world.yml

```diff
@@ -0,0 +1,73 @@
+name: Build Wasm component from Rust and push to OCI registry
+
+on: 
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+    
+    steps:
+     - name: Checkout repository
+       uses: actions/checkout@v4
+       with:
+        repository: ${{ github.repository_owner }}/${{ github.event.repository.name }}
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+     - name: Fetch repository content
+       run: |
+          git fetch origin
+          git checkout main
+     
+     - name: Setup wash CLI
+       uses: cosmonic-labs/setup-wash-action@main
+       with:
+         wash-version: latest
+
+     - name: Install wasm32-wasip2 target for Rust
+       run: |
+          rustup target add wasm32-wasip2
+
+     - name: Compile Wasm binary
+       working-directory: ./hello-world
+       run: |
+          wash build
+
+     - name: Login to GHCR
+       uses: docker/login-action@v3
+       with:
+         registry: ghcr.io
+         username: ${{ github.repository_owner }}
+         password: ${{ secrets.GITHUB_TOKEN }}
+
+     - name: Lowercase the organization name for ghcr.io
+       run: |
+          echo "GHCR_REPO_NAMESPACE=${GITHUB_REPOSITORY_OWNER,,}" >>${GITHUB_ENV}
+
+     - name: Push Wasm component to ghcr.io
+       working-directory: ./hello-world/target/wasm32-wasip2/debug/
+       run: |
+          wash oci push ghcr.io/${{ env.GHCR_REPO_NAMESPACE }}/components/hello-world:${{ github.ref_name }} ./hello_world.wasm
+
+     - name: Update image tag in Kubernetes manifest
+       working-directory: ./hello-world
+       run: |
+          DEPLOYMENT_FILE="manifests/component.yaml"
+          OLD_IMAGE=$(grep "image:" "$DEPLOYMENT_FILE" | awk '{print $2}')
+          NEW_IMAGE="ghcr.io/${{ env.GHCR_REPO_NAMESPACE }}/components/hello-world:${{ github.ref_name }}"
+
+          # Update the image tag
+          sed -i "s|image:.*|image: $NEW_IMAGE|" "$DEPLOYMENT_FILE"
+
+     - name: Commit and push manifest update
```

Comment:
Can you make this an automated PR instead (recommend the peter-evans/create-pull-request action)?